ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/sorter/BaseFeePendingTransactionsSorter.java

```diff
@@ -210,29 +203,41 @@ protected TransactionAddedStatus addTransaction(final TransactionInfo transactio
       LOG.trace("Adding {} to pending transactions", transactionInfo);
       pendingTransactions.put(transactionInfo.getHash(), transactionInfo);
 
-      if (pendingTransactions.size() > maxPendingTransactions) {
-        final Stream.Builder<TransactionInfo> removalCandidates = Stream.builder();
-        if (!prioritizedTransactionsDynamicRange.isEmpty())
-          lowestValueTxForRemovalBySender(prioritizedTransactionsDynamicRange)
-              .ifPresent(removalCandidates::add);
-        if (!prioritizedTransactionsStaticRange.isEmpty())
-          lowestValueTxForRemovalBySender(prioritizedTransactionsStaticRange)
-              .ifPresent(removalCandidates::add);
-
-        droppedTransaction =
-            removalCandidates
-                .build()
-                .min(
-                    Comparator.comparing(
-                        txInfo -> txInfo.getTransaction().getEffectivePriorityFeePerGas(baseFee)))
-                .map(TransactionInfo::getTransaction);
+      // check if this sender exceeds the transactions by sender limit:
+      var senderTxInfos = transactionsBySender.get(transactionInfo.getSender());
+      if (senderTxInfos.transactionCount() > poolConfig.getTxPoolMaxFutureTransactionByAccount()) {
+        droppedTransaction = senderTxInfos.maybeLastTx().map(TransactionInfo::getTransaction);
         droppedTransaction.ifPresent(
-            toRemove -> {
-              doRemoveTransaction(toRemove, false);
-              LOG.trace("Evicted {} due to transaction pool size", toRemove);
-            });
+            tx -> LOG.trace("Evicted {} due to too many transactions from sender", tx));
+
+      } else {
+        // else if we are over txpool limit, select the lowest value transaction to evict
+        if (pendingTransactions.size() > poolConfig.getTxPoolMaxSize()) {
+          final Stream.Builder<TransactionInfo> removalCandidates = Stream.builder();
+          if (!prioritizedTransactionsDynamicRange.isEmpty())
+            lowestValueTxForRemovalBySender(prioritizedTransactionsDynamicRange)
+                .ifPresent(removalCandidates::add);
+          if (!prioritizedTransactionsStaticRange.isEmpty())
+            lowestValueTxForRemovalBySender(prioritizedTransactionsStaticRange)
+                .ifPresent(removalCandidates::add);
+
+          droppedTransaction =
+              removalCandidates
+                  .build()
+                  .min(
+                      Comparator.comparing(
+                          txInfo -> txInfo.getTransaction().getEffectivePriorityFeePerGas(baseFee)))
+                  .map(TransactionInfo::getTransaction);
+        }
       }
+
+      droppedTransaction.ifPresent(
```

Comment:
Since we are calling addTransaction for each transaction we process (and synchronously), we should be able to continue to rely on the 1-in-1-out pattern (existing).

Though the periodic eviction thread for global nonce limits will behave the way you describe 👍